Modules/Relationships/src/Relationships.ConsumerApi/Controllers/RelationshipsController.cs

```diff
@@ -160,6 +176,13 @@ public async Task<IActionResult> CreateRelationship(CreateRelationshipCommand re
 
         return Ok(change);
     }
+
+    private async Task EnsurePeerIsValid(string peerIdentityAddress, CancellationToken cancellationToken)
```

Comment:
It's not that the peer is invalid or does not exist. It's just that it cannot accept the action that is being carried out because it is pending deletion. Perhaps this could be renamed to something like `EnsurePeerIsNotToBeDeleted`, as that would make it much clearer what's going on here.